OpenDreamRuntime/Procs/DMOpcodeHandlers.cs

```diff
@@ -128,6 +128,32 @@ static class DMOpcodeHandlers {
             return ProcStatus.Called;
         }
 
+        public static ProcStatus? CreateMultidimensionalList(DMProcState state)
+        {
+            var count = state.ReadInt();
+
+            List<DreamValue> values = new List<DreamValue>(count);
+            for (var i = 0; i < count; i++)
+            {
+                values.Add(state.Pop());
+            }
+            List<int> sizes = new List<int>(count);
+            foreach (var dv in values)
+            {
+                if (!dv.TryGetValueAsInteger(out var size))
+                {
+                    throw new Exception("Invalid list size");
+                }
+                sizes.Add(size);
+            }
```

Comment:
There's no need for two loops here. Non-integer sizes are also accepted and are treated as zero.
```suggestion
            List<int> sizes = new List<int>(count);
            for (var i = 0; i < count; i++)
            {
                state.Pop().TryGetValueAsInteger(out var size);
                sizes.Add(size);
            }
```